Legg til ratebegrenset ny Arenahistorikk-regel

perhenden-nav opened the pull request:
Legger til en ny regel for fordeling, og tar den i bruk i Dev men ikke i Prod. 

https://jira.adeo.no/browse/AAP-1672

Filteret på Arena-historikk beskrives best av kodedokumentasjonen på https://github.com/navikt/aap-arenaoppslag/blob/main/app/src/main/kotlin/arenaoppslag/aap/database/PersonRepository.kt#L246

Funksjonell beskrivelse er under arbeid på https://confluence.adeo.no/spaces/PAAP/pages/780350479/Søkere+med+Arena-historikk

perhenden-nav committed:
Legg til ratebegrenset ny Arenahistorikk-regel

AAP-1672

FredrikMeyer commented on flyt/src/test/kotlin/no/nav/aap/fordeler/regler/ArenaHistorikkRegelTest.kt:
```diff
@@ -14,42 +17,92 @@ import no.nav.aap.postmottak.journalpostogbehandling.journalpost.Brevkoder
 
     override fun harAapSakIArena(person: Person): PersonEksistererIAAPArena {
-        return PersonEksistererIAAPArena(true)
+        return if (person.identer().first().identifikator == identMedSignifikantSak) {
+            PersonEksistererIAAPArena(true)
+        } else {
+            PersonEksistererIAAPArena(false)
+        }
+    }
```
```suggestion
        return PersonEksistererIAAPArena(person.identer().first().identifikator == identMedSignifikantSak)
```

perhenden-nav replied:
En grei forbedring. Da det er i klassen som skal fjernes, lar jeg det bli.

perhenden-nav commented:
Når den prodsettes er det praktisk om den bruker Unleash feature-toggling, slik at det raskt kan skrus av og på. Kanskje også unleash for prosentverdien på inntak.

perhenden-nav committed:
Merge remote-tracking branch 'origin/main' into arena-historikk/ny_regel

perhenden-nav committed:
Bruk Unleash til å toggle ny arenahistorikk-regel

Oppdater grensesnitt brukt av alle regler sånn at det er mulig å gjøre dette

perhenden-nav committed:
Oppdater timeout pga ustabil test

perhenden-nav committed:
Legg til logg statement

perhenden-nav commented on flyt/src/main/kotlin/no/nav/aap/fordeler/regler/SignifikantArenaHistorikkRegel.kt:
```diff
@@ -16,7 +21,10 @@ class SignifikantArenaHistorikkRegel : Regel<SignifikantArenaHistorikkRegelInput
-        override val erAktiv = miljøConfig(prod = false, dev = true)
+        override fun erAktiv(gatewayProvider: GatewayProvider) = gatewayProvider.provide<UnleashGateway>().isEnabled(
+            PostmottakFeature.SignifikantHistorikkFraArena
+        )
```
Det er her det er party! (det er eneste plassen Unleash brukes)

perhenden-nav committed:
Logg ident til team-logs ikke vanlig log

FredrikMeyer reviewed (commented):
Se på commit feccdbfc (Greger), der ble det gjort en lignende endring, ved å sende inn Unleash i konstruktøren til regelen i stedet. Da slipper man å endre så mange filer.

:)

perhenden-nav commented:
> Se på commit [feccdbf](https://github.com/navikt/aap-postmottak-backend/commit/feccdbfc92160c420ba4b19c9df690a713b18f12) (Greger), der ble det gjort en lignende endring, ved å sende inn Unleash i konstruktøren til regelen i stedet. Da slipper man å endre så mange filer.
> 
En god grunn. Bra hukommelse. Jeg gjør det slik. Det kommer med noen minuser - det blir tre nivå av toggle av/på i regelen slik - erAktiv, ratebegrensning, og toggling separat fra erAktiv. 
Den duplikerte togglingen over alle regel-filene er et gode å bytte ut. 

Jeg tar det som to committer, så kan vi se hvordan det blir å bytte det ut på en mer komplett måte.

perhenden-nav committed:
med unleash class-var

perhenden-nav committed:
Revert "Bruk Unleash til å toggle ny arenahistorikk-regel"

This reverts commit 47498252c2e2f4ca79673f4db8cf294ef4e5bc9c.

perhenden-nav committed:
Endringer etter review

perhenden-nav committed:
Gjør erAktiv til en member function i Regel

Dette tillater oss å bruke ikke-statisk kode for å avgjøre om regelen er aktiv, feks. gateway eller repository.

perhenden-nav commented:
Gjort, de to siste comittene viser det, @FredrikMeyer

perhenden-nav committed:
Revert "Gjør erAktiv til en member function i Regel"

This reverts commit b930aa5f3d02cac60d5191ddf1145d6f8b4e356c.

perhenden-nav commented:
Det blir praktisk enklere å ta erAktiv endringen separat, så jeg dropper den fra denne PR-en.

perhenden-nav committed:
Støtt gradual rollout med Unleash

perhenden-nav committed:
Støtt gradual rollout med Unleash, 2/2

perhenden-nav committed:
Merge remote-tracking branch 'origin/main' into arena-historikk/ny_regel

etse commented on flyt/src/main/kotlin/no/nav/aap/fordeler/regler/SignifikantArenaHistorikkRegel.kt:
```diff
@@ -37,9 +49,15 @@ class SignifikantArenaHistorikkRegel : Regel<SignifikantArenaHistorikkRegelInput
     override fun generer(input: RegelInput): SignifikantArenaHistorikkRegelInput {
         val apiInternGateway = gatewayProvider.provide(AapInternApiGateway::class)
         val response = apiInternGateway.harSignifikantHistorikkIAAPArena(input.person, input.mottattDato)
+        secureLog.info(
```
Trenger vi å logge så mye sensitiv informasjon? Kunne vi kommet oss unna med å logge noe mindre sensitivt og droppet secure-logs? Tenker det er fint å om vi slipper secure-logs med mindre vi absolutt må ha det.

perhenden-nav replied:
Det er en uuid som logges, men det dekker også behovet med journalpostId, så jeg byttet til den, og også bort fra secure-logs 👍

etse commented on flyt/src/main/kotlin/no/nav/aap/fordeler/regler/Regel.kt:
```diff
@@ -23,7 +23,7 @@ class RegelMedInputgenerator<T>(val regel: Regel<T>, val inputGenerator: InputGe
 
 sealed interface RegelFactory<T> {
-    val erAktiv: Boolean
+    fun erAktiv(gatewayProvider: GatewayProvider): Boolean
```
Eneste som slår meg, ønker man faktisk å sende inn hele gateway-provideren her? Jeg tenker det er litt drit når man tester. Hva med å bare heller sende inn unleash-servicen? 🤔 Ikke sterke meninger om dette siden jeg ikke kjenner strukturen i denne appen så godt.

perhenden-nav replied:
Ja, gikk bort ifra den approachen, fant en bedre en

etse reviewed (approved):
Ser bra ut, ser veldig likt ut som andre apper. Og om du vil teste lokalt er det jo bare å kjøre opp med en faktisk reell unleash koblet mot dev og se hvordan det funker :)

perhenden-nav committed:
Unngå teamlogs, logg heller journalpostId
